packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -66,58 +80,56 @@ export const useGridKeyboardNavigation = (
   const navigateCells = React.useCallback(
     (params: GridCellParams, event: React.KeyboardEvent) => {
       event.preventDefault();
-
+      const { colIndex } = params;
       const key = mapKey(event);
       const isCtrlPressed = event.ctrlKey || event.metaKey || event.shiftKey;
-
       const cellEl = params.element!;
       cellEl.tabIndex = -1;
-
-      const currentColIndex = Number(cellEl.getAttribute('aria-colindex'));
-      const currentRowIndex = Number(cellEl.getAttribute('data-rowindex'));
+      const rowIndex =
```

Comment:
use params here. `params.rowIndex`

Reply:
It should be as it is because when applying a sorting the `rowIndex` is no longer the one you expect so the focus will start jumping between cells unpredictably.